test/common/http/conn_manager_impl_test_2.cc

```diff
@@ -520,8 +520,55 @@ TEST_F(HttpConnectionManagerImplTest, ConnectionDuration) {
   EXPECT_EQ(1U, stats_.named_.downstream_cx_max_duration_reached_.value());
 }
 
+TEST_F(HttpConnectionManagerImplTest, ConnectionDurationSafeHttp1) {
+  EXPECT_CALL(*codec_, protocol()).WillRepeatedly(Return(Protocol::Http10));
+  max_connection_duration_ = std::chrono::milliseconds(10);
+  Event::MockTimer* connection_duration_timer = setUpTimer();
+  EXPECT_CALL(*connection_duration_timer, enableTimer(_, _));
+  setup(SetupOpts().setHttp1SafeMaxConnectionDuration(true));
+
+  MockStreamDecoderFilter* filter = new NiceMock<MockStreamDecoderFilter>();
+  EXPECT_CALL(filter_factory_, createFilterChain(_))
+      .WillOnce(Invoke([&](FilterChainManager& manager) -> bool {
+        auto factory = createDecoderFilterFactoryCb(StreamDecoderFilterSharedPtr{filter});
+        manager.applyFilterFactoryCb({}, factory);
+        return true;
+      }));
+
+  EXPECT_CALL(*filter, decodeHeaders(_, false))
+      .WillOnce(Return(FilterHeadersStatus::StopIteration));
+  EXPECT_CALL(*filter, decodeData(_, true))
+      .WillOnce(Return(FilterDataStatus::StopIterationNoBuffer));
+  startRequest(true, "hello");
+
+  {
+    EXPECT_CALL(*connection_duration_timer, disableTimer());
+    EXPECT_LOG_CONTAINS(
```

Comment:
I would use the stats gauge that you have added, instead of watching for the log message. It will validate the observable behavior better.